src/functions/module_financial.ts

```diff
@@ -2341,6 +2350,83 @@ export const XIRR: AddFunctionDescription = {
   },
 };
 
+// -----------------------------------------------------------------------------
+// XNPV
+// -----------------------------------------------------------------------------
+export const XNPV: AddFunctionDescription = {
+  description: _lt("Net present value given to non-periodic cash flows.."),
+  args: args(`
+  discount (number) ${_lt("The discount rate of the investment over one period.")}
+  cashflow_amounts (number, range<number>) ${_lt(
+    "An range containing the income or payments associated with the investment."
+  )}
+  cashflow_dates (number, range<number>) ${_lt(
+    "An range with dates corresponding to the cash flows in cashflow_amounts."
+  )}
+  `),
+  returns: ["NUMBER"],
+  compute: function (
+    discount: PrimitiveArgValue,
+    cashflowAmounts: MatrixArgValue | PrimitiveArgValue,
+    cashflowDates: MatrixArgValue | PrimitiveArgValue,
+    rateGuess: PrimitiveArgValue = RATE_GUESS_DEFAULT
```

Comment:
not used ?

Reply:
That's just a magical argument that appeared one day in my head

Reply:
:open_mouth: